temporalcli/internal/printer/printer.go

```diff
@@ -0,0 +1,514 @@
+package printer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/olekukonko/tablewriter"
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/api/temporalproto"
+	"google.golang.org/protobuf/proto"
+)
+
+type Colorer func(string, ...interface{}) string
+
+type Printer struct {
```

Comment:
Seems like this might make more sense as an interface with JSON and Text implementations? A lot of the functions look to just be checking the `JSON` flag and returning early on a different path based on that.

Reply:
That's how I originally had it, but the abstraction was leaky and there'd only be two implementations. And callers often have to know whether they're using JSON or not anyways. So there was no value in extra layer of indirection, so it being right on the struct was fine (there is also a bit of shared code).